batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/internal/core/scanner/BatchArchiveDetector.java

```diff
@@ -250,4 +252,24 @@ public static IPackageFragmentRoot findPackageFragmentRoot(String jar, IJavaProj
 		return root;
 	}
 
+	boolean computeIsJarBatchArchive(File jarFile) {
+		try {
+			Indexer indexer = new Indexer();
+			Index index = JandexUtil.createJarIndex(jarFile, indexer);
```

Comment:
It is not stored. I found that for big jars it takes _longer_ to save/load than parse once more. For small jars it does not matter.